src/pipes/input-characters-validation.test.ts

```diff
@@ -0,0 +1,171 @@
+import { BadRequestException, InternalServerErrorException } from '@nestjs/common';
+
+import { InputCharacterValidationPipe } from './input-characters-validation';
+
+describe('InputCharacterValidationPipe', () => {
+  const queryMeta = { type: 'query' };
+  const paramMeta = { type: 'param' };
+  const flatParamMeta = { type: 'param', data: 'field2' };
+  const bodyMeta = { type: 'body' };
+  const unknownMeta = { type: '123' };
+  const urlDto = { field1: 'Test', field2: 'Test &%*?<>#/\\:' };
+  const bodyTextWithUnsupportedCharacters = 'Company ŽİĞİŞ A.Ş';
+  const bodyDto = {
+    field1: 'Test',
+    field2: {
+      field2_1: 'Test',
+      field2_2: 'Test',
+    },
+    field3: {
+      field3_1: {
+        field3_1_1: 'Test',
+      },
+      field3_2: {
+        field3_2_1: 'Test',
+        field3_2_2: 'Test',
+      },
+      field3_3: [
+        {
+          field3_3_1: 'Test',
+          field3_3_2: 'Test',
+        },
+        {
+          field3_3_1: 'Test',
+          field3_3_2: 'Test',
+        },
+      ],
+    },
+  };
+
+  let pipe: InputCharacterValidationPipe;
+
+  beforeEach(() => {
+    pipe = new InputCharacterValidationPipe();
+  });
+
+  describe('transform', () => {
+    it('throws an error if query param dto has unsupported characters', () => {
+      const functionToTest = () => {
+        pipe.transform(urlDto, queryMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+      expect(functionToTest).toThrow('Bad request');
+    });
+
+    it('throws an error if URL param dto has unsupported characters', () => {
+      const functionToTest = () => {
+        pipe.transform(urlDto, paramMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+      expect(functionToTest).toThrow('Bad request');
+    });
+
+    it('throws an error if URL flat param dto has unsupported characters', () => {
+      const functionToTest = () => {
+        pipe.transform(urlDto['field2'], flatParamMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+      expect(functionToTest).toThrow('Bad request');
+    });
```

Comment:
It'd be good to also test that using valid url params returns the original value without throwing